torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2181,6 +2181,27 @@ def sample_inputs_max_min_binary(op_info, device, dtype, requires_grad, **kwargs
                   for input_tensor, other_tensor in args_for_binary_op)
     return inputs
 
+def sample_inputs_normalize(self, device, dtype, requires_grad, **kwargs):
+    make_arg = partial(make_tensor, low=-1, high=1, device=device, dtype=dtype, requires_grad=requires_grad)
+
+    # Order: input_shape, p (exponent), dim 
+    cases = (((2, 1, 4, 5), 1., 2),
+             ((2, 3, 4, 5), 2., 1),
+             ((1, 2, 4, 5), 0.5, 0),
+             ((1, 3, 4, 5), -1., 1),
+             ((1, 3, 4, 5), 0., -1),
+             ((), 1.2, 0))
+
+    def generator():
+        for input_shape, p, dim in cases:
+            yield SampleInput(make_arg(input_shape), args=(p, dim))
+        # No args case
+        yield SampleInput(make_arg((2, 3, 4, 5)))
```

Comment:
These are good extra cases; a simpler way of writing this in the future might just be to add a dict to the tuples in `cases` above that was the kwargs. Most of them would be empty but that seems OK

Reply:
Thanks, @mruberry for taking a look. I've made the changes in the recent commit, I hope I didn't get your suggestion wrong? I'm sorry if I did.

One extra point, I've realized - since these aren't kw-only args, hence I would like to check their positions as well (`p` is floating-point and `dim` is integral type). So I've added a separate extra case while keeping `kwargs` in the `cases` tuple as a `dict`.

Do you think that's fine?

Reply:
I appreciate your thinking and you could structure the cases like this:

```
cases = (
  (input, (args), {kwargs})
)
```

To cover both positional and keyword argument cases. Testing that a function accepts positional and keyword args correctly may not be that interesting, though, since it's testing the arg parser more than the function's logic. It might catch accidental regressions where an operation was made kwarg-only by mistake, but that seems like a rare case.

So I think the additional case is fine, but it can be modeled alongside the other cases, and it's probably not necessary. Does that make sense?

Reply:
Thanks, @mruberry! I agree, the positional arg testing isn't that interesting here. I removed this from the cases here, testing through `kwargs` should be enough here, I suppose.